services/web-server/test/session_store_test.js

```diff
@@ -67,6 +68,28 @@ helper.secrets.mockSuite(testing.suiteName(), [], function(mock, skipping) {
 
       assert.ok(val, 'entry exists');
     });
+    test('it shouldn\'t get the specified session when only the hash matches', async function () {
+      const store = getStore();
+
+      const currentSession = 'sessionid1';
+      const currentHash = hash(currentSession);
+
+      const newSession = 'sessionid2';
+      const newHash = hash(newSession);
+
+      //creates a new session
+      await store.set(currentSession, { cookie: {} });
+
+      // updates the hash of that session
+      await helper.db._withDbClient(async client => {
```

Comment:
and here, I think you want `_withClient` :)

Reply:
(also, it needs a first argument of 'read' or, in this case 'write' -- that's another thing that #4280 makes easier)